### PR TITLE
feat: Enroll drop revoked clients

### LINKED ISSUE #1927: When an enrollment is revoked, any clients connected via that enrollment should be immediately disconnected
### Is your feature request related to a problem? Please describe.

Without this safeguard, a revoked app could remain connected until the network connection is lost or the server is restarted etc

### Describe the solution you'd like

When an enrollment is revoked, any clients connected via that enrollment should be immediately disconnected

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by murali-shris**
Closes https://github.com/atsign-foundation/at_server/issues/1927
**- What I did**
- When an enrollment is revoked and if that enrollment has any active client, close the connection to the client on the server
**- How I did it**
- Implemented a method _dropRevokedClientConnection which drop connection on the server when enrollment is revoked
    - If the connection to drop is not the current connection, directly close the connection without any warning
    - if the connection to drop is current connection and force flag is set..send a message to client in the response and drop the connection after 10 seconds.
- Modified enroll verb functional tests to check for inbound connection count before and after enroll:revoke 
**- How to verify it**
- unit and functional tests should pass

### CODE REVIEW

**Comment by murali-shris** [comment]
Resolved the functional test failure for enroll:revoke:force
working on functional test for revoking enrollment and verify using stats verb whether connection got closed

File: packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart [hunk 1]
```diff
@@ -331,6 +338,43 @@ class EnrollVerbHandler extends AbstractVerbHandler {
     responseJson['enrollmentId'] = enrollmentIdFromParams;
   }
 
+  Future<void> _dropRevokedClientConnection(
+      String enrollmentId,
+      String? forceFlag,
```
Comment (gkc) [inline packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart]: boolean instead

File: packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart [hunk 2]
```diff
@@ -104,6 +105,12 @@ class EnrollVerbHandler extends AbstractVerbHandler {
           }
           await _handleEnrollmentPermissions(enrollVerbParams, currentAtSign,
               operation, responseJson, response);
+          if (responseJson['status'] == EnrollmentStatus.revoked.name) {
+            logger.finer(
+                'Dropping connection for enrollmentId: $enrollmentIdFromParams');
+            await _dropRevokedClientConnection(
+                enrollmentIdFromParams!, forceFlag, atConnection, responseJson);
```
Comment (gkc) [inline packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart]: Suggest pass forceFlag as a boolean i.e. (forceFlag != null)

File: packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart (hunk 1 above)
Comment (murali-shris) [reply packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart]: done

File: packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart (hunk 2 above)
Comment (murali-shris) [reply packages/at_secondary_server/lib/src/verb/handler/enroll_verb_handler.dart]: done

**Review by gkc — APPROVED** [review]

**Pull request merged**